src/app/beer_garden/router.py

```diff
@@ -293,13 +293,36 @@ def execute_local(operation: Operation):
 
 
 def update_api_heartbeat(operation: Operation):
+
     if (
-        operation.source_garden_name is not None
-        and operation.source_garden_name != config.get("garden.name")
-        and operation.source_api is not None
+        operation.source_api is not None
         and operation.operation_type == "PUBLISH_EVENT"
         and operation.model.name == Events.GARDEN_SYNC.name
     ):
+        if operation.source_garden_name == config.get("garden.name"):
+
+            if operation.model.payload.name != operation.source_garden_name:
+
+                local_garden = get_garden(config.get("garden.name"))
+
+                # Will only support mapping 1 hop away legacy Garden Syncs
+                not_child_garden = True
+                for child in local_garden.children:
+                    if child.name == operation.model.payload.name:
+                        logger.warning(
+                            (
+                                "Legacy (3.16 or prior) GARDEN_SYNC operation "
+                                f"seen for Beer-Garden '{operation.model.payload.name}'"
+                            )
+                        )
+                        operation.source_garden_name = operation.model.payload.name
+                        not_child_garden = False
```

Comment:
The not makes this a bit hard to read. Did you consider child_garden = False/True and if not child_garden?